fastlane/Helpers/Match.swift

```diff
@@ -33,6 +33,51 @@ enum Match {
                 force: .userDefined(isForce)
             )
         }
+        updateCodeSigning(type: type, appIdentifier: appIdentifier)
+    }
+    
+    static func updateCodeSigning(type: MatchType, appIdentifier: [String]) {
+        // Update Code signing from automatic to manual
+        switch type {
+        case .development:
+            updateCodeSigningSettings(
+                path: Constant.projectPath,
+                useAutomaticSigning: .userDefined(false),
+                targets: .userDefined([Constant.projectName]),
+                buildConfigurations: .userDefined(["Release Staging"]),
+                codeSignIdentity: .userDefined("iPhone Developer"),
+                profileName: .userDefined("match AdHoc \(Constant.stagingBundleId)")
+            )
+        case .adHoc:
+            if appIdentifier.first == Constant.productionBundleId {
+                updateCodeSigningSettings(
+                    path: Constant.projectPath,
+                    useAutomaticSigning: .userDefined(false),
+                    targets: .userDefined([Constant.projectName]),
+                    buildConfigurations: .userDefined(["Debug Production"]),
+                    codeSignIdentity: .userDefined("iPhone Distribution"),
+                    profileName: .userDefined("match AdHoc \(Constant.productionBundleId)")
+                )
+            } else {
+                updateCodeSigningSettings(
+                    path: Constant.projectPath,
+                    useAutomaticSigning: .userDefined(false),
+                    targets: .userDefined([Constant.projectName]),
+                    buildConfigurations: .userDefined(["Release Staging"]),
+                    codeSignIdentity: .userDefined("iPhone Distribution"),
+                    profileName: .userDefined("match AdHoc \(Constant.stagingBundleId)")
+                )
+            }
+        case .appStore:
+            updateCodeSigningSettings(
+                path: Constant.projectPath,
+                useAutomaticSigning: .userDefined(false),
+                targets: .userDefined([Constant.projectName]),
+                buildConfigurations: .userDefined(["Release Production"]),
+                codeSignIdentity: .userDefined("iPhone Distribution"),
+                profileName: .userDefined("match AppStore \(Constant.productionBundleId)")
```

Comment:
I prefer to define `AdHoc` and `AppStore` as `MatchType`'s attributes such as `type`:
```swift
    enum MatchType: String {
        ...
        var type: String {
            switch self {
            case .development: return "Development"
            case .adHoc: return "AdHoc"
            case .appStore: return "AppStore"
            }
        }
    }
```

And a private function to make `profileName` so we don't duplicate many `match` words.